pythonforandroid/archs.py

```diff
@@ -56,7 +62,13 @@ def get_env(self, with_flags_in_cc=True, clang=False):
             cflags.append('-gcc-toolchain {}'.format(toolchain))
 
         env['CFLAGS'] = ' '.join(cflags)
-        env['LDFLAGS'] = ' '
+
+        # Link the extra global link paths first before anything else
+        # (such that overriding system libraries with them is possible)
+        env['LDFLAGS'] = ' ' + " ".join([
+            "-L'" + l.replace("'", "'\"'\"'") + "'"
```

Comment:
I'm not completely clear, what is this fixing?

Reply:
@inclement this is just adding these new global link options as proposed, and adding them first so they take precedence. (actually now that you say it I'm not sure the `-L` order matters so maybe the comment is nonsense and it could have been added at another point in that function, but then again I still don't see a reason to move it somewhere else) The replace is there because it should be, it does proper quoting in all cases for paths including space, single quotes, ...

Reply:
I think the -L order does matter, just wasn't clear what the effect of the escaping is. I'm happy to believe it does the right thing, but maybe it could go in a different `quote_path` function to be clear?

Reply:
@inclement this escaping is the ultimate(TM) shell escaping which should be used *everywhere* - if it was, then we'd have no issue with spaces in folders and all these other things.

It's pretty simple as well, if you think about it: single quotes basically disable everything that could do anything (`$`, space chars, ...) - except for terminating single quotes. So the single quotes themselves are escaped by `'"'"'`, spaced out this is: `'  "  '  "  '`. Explanation: it terminates the current single quote block, then adds a single quote in double escapes (to avoid starting a new block), then starts a new single quote block directly attached.

I got this escaping method from `shlex.quote` actually, I believe?! I forgot. Maybe we could actually use that one, I tend to forget it exists. Anyway, my point was I didn't make this up myself but smarter people did, so the escaping method should be solid :smile:

Reply:
Sure, I understand what it's doing and I totally believe it's correct, it's just hard to judge it without knowing the rules of quoting offhand. It sounds like using shlex.quote would be the best option.

Reply:
@inclement you're right, it should just be `shlex.quote` :+1: